internal/store/horizontalpodautoscaler.go

```diff
@@ -177,6 +178,69 @@ func hpaMetricFamilies(allowLabelsList []string) []generator.FamilyGenerator {
 				}
 			}),
 		),
+		*generator.NewFamilyGenerator(
+			"kube_horizontalpodautoscaler_status_current_metric",
+			"The metric current status used by this autoscaler when calculating the desired replica count.",
+			metric.Gauge,
+			"",
+			wrapHPAFunc(func(a *autoscaling.HorizontalPodAutoscaler) *metric.Family {
+				ms := make([]*metric.Metric, 0, len(a.Status.CurrentMetrics))
+				for _, m := range a.Status.CurrentMetrics {
```

Comment:
Do you think this could be extracted into a function that's called from both here and `kube_horizontalpodautoscaler_spec_target_metric`?